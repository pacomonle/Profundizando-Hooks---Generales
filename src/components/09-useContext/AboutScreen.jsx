import React, { useContext } from 'react'
import { UserContext } from './UserContext'

const AboutScreen = () => {
    
        const {user} = useContext(UserContext)
        console.log(user)
   
       return (
           <div>
               <h1>About</h1>
               <hr />
               <pre className='container'>
               {
                   JSON.stringify(user, null ,3)
               }
               </pre>
           </div>
       )
    
}

export default AboutScreen
