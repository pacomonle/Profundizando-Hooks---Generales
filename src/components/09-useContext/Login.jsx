import React, { useContext } from 'react'
import { UserContext } from './UserContext'

const Login = () => {

const {setuser} = useContext(UserContext)


    return (
        <div>
            <h1>Login</h1>
            <hr />
            <button 
            className='btn btn-primary'
            onClick={()=>{setuser({
                id:123,
                name: 'Monleon',
                email: 'prueba@prueba.com'
            })}}
            type="button">
            Login
            </button>
            <button 
            className='btn btn-danger mx-4'
            onClick={()=>{setuser({})}}
            type="button">
            Logout
            </button>
        </div>
    )
}

export default Login
