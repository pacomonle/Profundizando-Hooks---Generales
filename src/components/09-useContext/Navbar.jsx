import React from 'react'
import { Link, NavLink } from 'react-router-dom'

const Navbar = () => {
    return (
    <>
        <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <Link to='/' className="navbar-brand">useContext app</Link>
            
            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                <NavLink to='/home' exact activeClassName='active' className="nav-link" >Home</NavLink>
                <NavLink to='/about' exact activeClassName='active' className="nav-link" >About</NavLink>
                <NavLink to='/login' exact activeClassName='active' className="nav-link" >Login</NavLink>
                </div>
            </div>
        </nav>
     </>
    )
}

export default Navbar
