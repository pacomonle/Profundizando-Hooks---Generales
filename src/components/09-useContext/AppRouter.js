// componente que se encarga de organizar las rutas
import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route, 
    Redirect
  } from "react-router-dom";
import AboutScreen from './AboutScreen';
import HomeScreen from './HomeScreen';
import Login from './Login';
import Navbar from './Navbar';


const AppRouter = () => {
    return (
        <Router>
            <div>
                {/*NAVBAR*/}
                <Navbar></Navbar>
                {/*RUTAS*/}
                <Switch>
                    <Route exact path='/' component={HomeScreen} />
                    <Route exact path='/home' component={HomeScreen} />
                    <Route exact path='/about' component={AboutScreen} />
                    <Route exact path='/login' component={Login}/>
                    {/*RUTA 404
                    <Route component={HomeScreen} />
                    */}
                    <Redirect to='/' />
                </Switch>
            </div>
        </Router>
        
    )
}

export default AppRouter
