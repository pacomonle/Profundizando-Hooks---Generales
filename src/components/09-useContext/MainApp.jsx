import React from 'react'
import { useState } from 'react'
import '../02-useEffect/effects.css'
import AppRouter from './AppRouter'
import { UserContext } from './UserContext'

const MainApp = () => {

    const [user, setuser] = useState({})
    

    return (
        <UserContext.Provider value={{
            user,
            setuser
        }}>
          <AppRouter></AppRouter>
        </UserContext.Provider>   
    )
}

export default MainApp
