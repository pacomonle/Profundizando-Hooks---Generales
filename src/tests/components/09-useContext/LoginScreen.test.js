import React from 'react';
import { mount } from 'enzyme';
import { UserContext } from '../../../components/09-useContext/UserContext';
import  Login  from '../../../components/09-useContext/Login';


describe('Pruebas en <LoginScreen />', () => {
// setuser es una funcion , para simularla necesitamos el jest.fn()
    const setuser = jest.fn();
    
    const wrapper = mount(
        <UserContext.Provider value={{
            setuser
        }}>
            <Login />
        </UserContext.Provider>
    )

    test('debe de mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });


    test('debe de ejecutar el setUser con el argumento esperado', () => {
        // otra opcion para simular la funcion es el simulate('click')
       //  esto da una funcion por eso () para ejecutarla
        wrapper.find('button').at(0).prop('onClick')();

        expect( setuser ).toHaveBeenCalledWith({
            id:123,
            name: 'Monleon',
            email: 'prueba@prueba.com'
        })
        
    });
    
    

})
