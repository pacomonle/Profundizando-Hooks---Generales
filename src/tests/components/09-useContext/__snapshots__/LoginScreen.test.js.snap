// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    Login
  </button>
  <button
    className="btn btn-danger mx-4"
    onClick={[Function]}
    type="button"
  >
    Logout
  </button>
</div>
`;
